### ISSUE TITLE
Frequent False ANT+ Cadences in Train Mode

### ISSUE THREAD

**Issue by pschluet** [GoldenCheetah/GoldenCheetah#4099]
### Description of Issue
I'm having issues in train mode where, if I'm pedaling above 80 RPM, my cadence frequently shows false cadence readings of 80 RPM (I know it's incorrect because it jumps back down to 80, then back up to my real cadence while my pedaling remains consistent; I confirmed my real cadence via a separate bike computer). Here's a plot from one of my rides showing the frequent jumps down to 80 RPM.

<img width="1454" alt="Screen Shot 2021-12-05 at 6 21 16 PM" src="https://user-images.githubusercontent.com/16847359/144770734-cbed3338-8bad-412b-a01f-8b32785779d3.png">

### Fix
PR #4098 fixes this issue. I tested the fix on my trainer, and it worked.

### Equipment
All connected via ANT+:
- Kickr Core trainer
- Polar H10 heart rate monitor
- Garmin Cadence Sensor 2

### Data Files
- [antlog.raw](https://github.com/GoldenCheetah/GoldenCheetah/files/7657120/antlog.txt) (change the file extension to .raw; github wouldn't accept a .raw, so I changed extension to .txt)
- [2021_12_04_07_17_36.json](https://github.com/GoldenCheetah/GoldenCheetah/files/7657123/2021_12_04_07_17_36.txt) (again, change file extension to .json when you download it)

### Version Information
![image](https://user-images.githubusercontent.com/16847359/144770636-d2f3512d-2c38-44f3-acec-95cc9b071794.png)

**Comment by liversedge** [comment]
This is not an appropriate fix for your issue. The source of data fluctuations are almost certainly due to the devices you have connected. This is probably best raised in the forum.

**Comment by pschluet** [comment]
@liversedge I'm pretty sure there's an issue in the code and [my PR](#4098) fixes it. I looked in the raw ANT+ data and confirmed that, based on the ANT+ messages I was getting, the code that calculates RPM would calculate the same incorrect values that I was seeing. I tested my fix on the real hardware, and it fixed the problem.

I'm using one of the most common cadence sensors out there (Garmin Cadence Sensor 2), so I'm thinking that would reduce the likelihood that there's an issue with the sensor. Additionally, the cadence values read correctly when being received on my Coros Pace 2 watch.
